x/margin/keeper/msg_server_test.go

```diff
@@ -387,6 +430,239 @@ func TestKeeper_Close(t *testing.T) {
+				MtpAddress: "sif1azpar20ck9lpys89r8x7zc8yu0qzgvtp48ng5v",
+				Id:         1,
+			},
+			msgOpen: types.MsgOpen{
+				CollateralAsset: "rowan",
+				BorrowAsset:     "xxx",
+				Position:        types.Position_SHORT,
+			},
+			poolAsset:     "xxx",
+			token:         "xxx",
+			poolEnabled:   true,
+			fundedAccount: true,
+			errString:     errors.New("SHORT: mtp position invalid"),
+		},
+	}
+
+	for _, tt := range table {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, app := test.CreateTestAppMargin(false)
+			marginKeeper := app.MarginKeeper
+
+			app.TokenRegistryKeeper.SetToken(ctx, &tokenregistrytypes.RegistryEntry{
+				Denom:       tt.token,
+				Decimals:    18,
+				Permissions: []tokenregistrytypes.Permission{tokenregistrytypes.Permission_CLP},
+			})
+
+			msgServer := keeper.NewMsgServerImpl(marginKeeper)
+
+			asset := clptypes.Asset{Symbol: tt.poolAsset}
+			pool := clptypes.Pool{
+				ExternalAsset:        &asset,
+				NativeAssetBalance:   sdk.NewUint(1000000000),
+				NativeLiabilities:    sdk.NewUint(1000000000),
+				ExternalCustody:      sdk.NewUint(1000000000),
+				ExternalAssetBalance: sdk.NewUint(1000000000),
+				ExternalLiabilities:  sdk.NewUint(1000000000),
+				NativeCustody:        sdk.NewUint(1000000000),
+				PoolUnits:            sdk.NewUint(1),
+				Health:               sdk.NewDec(1),
+			}
+			if tt.poolEnabled {
+				marginKeeper.SetEnabledPools(ctx, []string{tt.poolAsset})
+			}
+
+			marginKeeper.ClpKeeper().SetPool(ctx, &pool)
+
+			var address string
+
+			if tt.fundedAccount {
+				nativeAsset := tt.msgOpen.CollateralAsset
+				externalAsset := clptypes.Asset{Symbol: tt.msgOpen.BorrowAsset}
+
+				nativeCoin := sdk.NewCoin(nativeAsset, sdk.Int(sdk.NewUint(1000000000000)))
+				externalCoin := sdk.NewCoin(externalAsset.Symbol, sdk.Int(sdk.NewUint(1000000000000)))
+				err := app.BankKeeper.MintCoins(ctx, clptypes.ModuleName, sdk.NewCoins(nativeCoin, externalCoin))
+				require.Nil(t, err)
+
+				nativeCoin = sdk.NewCoin(nativeAsset, sdk.Int(sdk.NewUint(10000)))
+				externalCoin = sdk.NewCoin(externalAsset.Symbol, sdk.Int(sdk.NewUint(10000)))
+
+				_signer := clptest.GenerateAddress(clptest.AddressKey1)
+				address = _signer.String()
+				err = sifapp.AddCoinsToAccount(types.ModuleName, app.BankKeeper, ctx, _signer, sdk.NewCoins(nativeCoin))
+				require.Nil(t, err)
+				marginKeeper.BankKeeper().SendCoinsFromAccountToModule(ctx, _signer, types.ModuleName, sdk.NewCoins(nativeCoin))
+			} else {
+				address = tt.msgForceClose.Signer
+			}
+
+			msg := tt.msgForceClose
+			msg.Signer = address
+
+			var signer string = msg.Signer
+			if tt.overrideSigner != "" {
+				signer = tt.overrideSigner
+			}
+
+			if !tt.mtpCreateDisabled {
+				addMTPKey(t, ctx, app, marginKeeper, tt.msgOpen.CollateralAsset, tt.msgOpen.BorrowAsset, signer, tt.msgOpen.Position, 1)
+			}
+
+			_, got := msgServer.ForceClose(sdk.WrapSDKContext(ctx), &msg)
```

Comment:
@canercandan the MTP that is closed uses msg.MtpAddress, not msg.Signer, so the above overrides need to apply to MtpAddress as well.

Reply:
@timlind thanks for your review, I fixed this issue in the last commit I pushed and added additional test cases with different values for force close threshold, things are working as expected.

I did notice though that the force close threshold param is not initiated here: https://github.com/Sifchain/sifnode/blob/feature/margin-1-multiple-positions/x/margin/types/genesis.go#L17

cc @AustinoBombino